modules/core/src/test/java/org/apache/ignite/testframework/TestDependencyResolver.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.testframework;
+
+import org.apache.ignite.internal.processors.resource.DependencyResolver;
+import org.apache.ignite.internal.processors.resource.NoopDependencyResolver;
+
+/**
+ * Test implementation of dependency resolver. You can pass your logic via method reference.
+ */
+public class TestDependencyResolver extends AbstractTestDependencyResolver {
```

Comment:
Can we remove this class?
We can pass dependency resolver itself by method reference like you did in IgniteClientCacheInitializationFailTest, for example.